tests/unit/test_federation_policy

```diff
@@ -1,16 +1,96 @@
 #!/bin/bash
 
+# Requirements:
+# - cf-agent installed
+
 echo "1..4"
 
 function assert() {
-    grep $1 $2 || {
+    grep "$1" "$2" || {
         echo "error: did not find '$1' in file: $2"
         exit 1
     }
 }
 
 set -e
 
+# borrow PATH setup from nova/tests/reporting/common.sh
+[ -z "$CFE_ROOT" ] && CFE_ROOT=$(cd ../../../ && pwd)
+PATH="$CFE_ROOT/core/cf-agent/:$PATH"
+
+# === SETUP FAKE ENVIRONMENT in /tmp for CFE_WORKDIR and built binaries/masterfiles ===
+# TODO: reuse much of this
+CFE_WORKDIR=$(mktemp -d)
+TMPLOG=$(mktemp)
+
+# for debugging
+#set +e
+#find /opt/cfengine > opt-cfengine-before.log
+#find "$CFE_ROOT" > cfe-root-before.log
+#set -e
+
+function tear_down() {
+    # for debugging
+    #set +e
+    #find /opt/cfengine > opt-cfengine-after.log
+    #find "$CFE_ROOT" > cfe-root-after.log
+    #diff opt-cfengine-before.log opt-cfengine-after.log
+    #diff cfe-root-before.log cfe-root-after.log
+    rm -rf "$CFE_WORKDIR"
+    rm "$TMPLOG"
+    sudo rm -rf /opt/cfengine
+
+    # re-make masterfiles so that the shebang in modules/packages/apt_get
+    # is reverted to /var/cfengine/bin/python from our tmp dir
+    pushd ../../
+    ./configure
+    make
+    popd
+
+    set -e
+}
+
+trap tear_down SIGTERM
+trap tear_down EXIT
+trap tear_down SIGINT
+
+mkdir -p "$CFE_WORKDIR/bin"
+mkdir -p "$CFE_WORKDIR/lib"
+mkdir -p "$CFE_WORKDIR/state"
+
+# NOTE: environment in jenkins/travis guarantees these are already built
+pushd "$CFE_ROOT/core"
+if [ ! -f configure ]; then
+  ./autogen.sh
+  make -j4
+fi
+popd
+
+pushd "$CFE_ROOT/enterprise"
+if [ ! -f configure ]; then
+  ./autogen.sh
+  make -j4
+fi
+popd
+
+cp "$CFE_ROOT/enterprise/enterprise-plugin/.libs/cfengine-enterprise.so" "$CFE_WORKDIR/lib/"
+cp "$CFE_ROOT/core/cf-promises/cf-promises" "$CFE_WORKDIR/bin"
+cp -R "$CFE_ROOT/core/cf-promises/.libs" "$CFE_WORKDIR/bin"
+
+pushd ../../
+./configure --prefix "$CFE_WORKDIR"
+make install
+popd
+
+cp -R "$CFE_WORKDIR/masterfiles" "$CFE_WORKDIR/inputs"
+
+export CFENGINE_TEST_OVERRIDE_WORKDIR="$CFE_WORKDIR"
```

Comment:
This shouldn't be needed if you specify `--prefix "$CFE_WORKDIR"` also when running `./autogen.sh` in *core* and *enterprise* above.

Reply:
It seems things like check for cf-promises is very much hard coded to /var/cfengine/bin so I will leave this as-is unless you have another idea?

Reply:
Hmm, that's weird, I remember fixing it to use `GetBinDir()`.

Reply:
And `GetBinDir()` should respect `--prefix=/something/special` from *configure* (and thus *autogen.sh*).

Reply:
Yeah, I was thinking about this more and realized it should work. Maybe the jenkins run was using the wrong binary. I'll look more.